### ISSUE TITLE
Less number of components after re-uploading BOM to existing project

### ISSUE THREAD

**Issue by software-testing-professional** [DependencyTrack/dependency-track#1724]
### Current Behavior:
If a new project is created from an uploaded BOM, the number of components is correct. 

But if the same file is re-uploaded (regardless via UI or API), the number of components decreases. 
For example from 1922 to 1806. 

logs: 
```
2022-06-16 20:40:33,340 INFO [BomUploadProcessingTask] Processing CycloneDX dependency graph for project: 89e25caf-1334-4a68-bc64-93bb5fe0e5c0
2022-06-16 20:40:33,501 INFO [BomUploadProcessingTask] Processed 1922 components and 0 services uploaded to project 89e25caf-1334-4a68-bc64-93bb5fe0e5c0
2022-06-16 20:40:40,608 INFO [InternalAnalysisTask] Starting internal analysis task
2022-06-16 20:40:50,816 INFO [InternalAnalysisTask] Internal analysis complete
2022-06-16 20:40:50,824 INFO [OssIndexAnalysisTask] Starting Sonatype OSS Index analysis task
2022-06-16 20:41:02,187 INFO [OssIndexAnalysisTask] Sonatype OSS Index analysis complete
2022-06-16 20:41:02,189 INFO [PolicyEngine] Evaluating 1922 component(s) against applicable policies
```

UI: 
![image](https://user-images.githubusercontent.com/82155716/174161859-1e45131b-408d-455e-a1dd-34bbdb1c9f04.png)
1808 components
If this inventory is downloaded via UI, it contains 1808 components. 

Each time I repeat the re-upload (initial file from disk), the number of components in the UI is different. (1808, 1922, 1602 etc.).

So:
INITIAL upload of BOM file (new project created during upload): # of components is correct
RE-UPLOAD of initial bom file (project already exists): # of components is NOT correct / varies

The imported BOM was created via `cyclonedx-cli merge ...` from several individual BOMs. 
These BOMs were downloaded from Dependency-Track projects previously. 

### Steps to Reproduce:
Merge two or more cdx files with cyclonedx-cli, and upload them to Dependency-Track. 

### Expected Behavior:
The number of components is consistent. 

### Environment:
  
  - Dependency-Track Version: 4.4.2
  - Distribution:  Docker
  - BOM Format & Version: JSON 1.4 // XML 1.4
  - Database Server: [ PostgreSQL ]
  - Browser: Chrome

### Additional Details:

If I get a GO from my project, I will attach a CDX JSON file for reproducing this behavior.

**Comment by software-testing-professional** [comment]
What I also checked after merging several files into one BOM: 
```
$ cyclonedx-cli validate --input-file bom/final.bom.cdx.json 
BOM validated successfully.
```

**Comment by nscuro** [comment]
Hi @software-testing-professional, thanks for reporting! 👋

Unfortunately I can't reproduce this. Tested with individual and merged SBOMs, but I don't see the behavior you're describing.

For reference, I tested with these SBOMs (in addition to Dependency-Track's own SBOM):

* https://raw.githubusercontent.com/CycloneDX/bom-examples/master/SBOM/juice-shop-11.1.2/bom.json
* https://github.com/CycloneDX/bom-examples/blob/master/SBOM/keycloak-10.0.2/bom.json

When merged, they add up to 1743 components. This number stayed consistent throughout multiple re-uploads.

```shell
$ cyclonedx --version
0.24.0

$ cyclonedx merge --input-files ~/Downloads/keycloak.bom.json ~/Downloads/juice-shop.bom.json --output-file ~/Downloads/merged.bom.json --output-format json
Processing input file /Users/nscur0/Downloads/keycloak.bom.json
    Contains 903 components
Processing input file /Users/nscur0/Downloads/juice-shop.bom.json
    Contains 840 components
Writing output file...
    Total 1743 components
```

If you do the same with these SBOMs on your DT instance, can you replicate the behavior you've been seeing earlier?
If not, can you share the SBOM that triggers the issue for you?

**Comment by software-testing-professional** [comment]
Hi @nscuro 

thank you for your support. :-)
I will test again with your BOM files, and report asap. 

Regards, Michael

**Comment by software-testing-professional** [comment]
Finally, I think I found the cause for this. 

The CycloneDX BOM that is imported into DependencyTrack has for example these two components: 

```
[
  {
    "type": "library",
    "name": "alpine-baselayout-data",
    "version": "3.2.0-r22",
    "description": "Alpine base dir structure and init scripts",
    "licenses": [
      {
        "license": {
          "id": "GPL-2.0-only"
        }
      }
    ],
    "cpe": "cpe:2.3:a:alpine-baselayout-data:alpine-baselayout-data:3.2.0-r22:*:*:*:*:*:*:*",
    "purl": "pkg:alpine/alpine-baselayout-data@3.2.0-r22?arch=x86_64&distro=alpine-3.16.1&upstream=alpine-baselayout",
    "externalReferences": [
      {
        "url": "https://git.alpinelinux.org/cgit/aports/tree/main/alpine-baselayout",
        "type": "distribution"
      }
    ]
  },
  {
    "type": "library",
    "name": "alpine-baselayout-data",
    "version": "3.2.0-r22",
    "description": "Alpine base dir structure and init scripts",
    "licenses": [
      {
        "license": {
          "id": "GPL-2.0-only"
        }
      }
    ],
    "cpe": "cpe:2.3:a:alpine-baselayout-data:alpine-baselayout-data:3.2.0-r22:*:*:*:*:*:*:*",
    "purl": "pkg:alpine/alpine-baselayout-data@3.2.0-r22?arch=x86_64&distro=alpine-3.16.2&upstream=alpine-baselayout",
    "externalReferences": [
      {
        "url": "https://git.alpinelinux.org/cgit/aports/tree/main/alpine-baselayout",
        "type": "distribution"
      }
    ]
  }
]
```

But DependencyTrack  shows this component `alpine-baselayout-data` only once:

![image](https://user-images.githubusercontent.com/82155716/189062080-076b851b-b2b5-4910-a27a-61713a6206e9.png)

![image](https://user-images.githubusercontent.com/82155716/189064127-18587b68-1ae6-426d-814e-d079751a0352.png)


Looking at the imported CycloneDX data, both components have the same CPE
`"cpe": "cpe:2.3:a:alpine-baselayout-data:alpine-baselayout-data:3.2.0-r22:*:*:*:*:*:*:*"`

but different PURLs
`"purl": "pkg:alpine/alpine-baselayout-data@3.2.0-r22?arch=x86_64&distro=alpine-3.16.1&upstream=alpine-baselayout"` 
vs.
`"purl": "pkg:alpine/alpine-baselayout-data@3.2.0-r22?arch=x86_64&distro=alpine-3.16.2&upstream=alpine-baselayout"`

**Comment by nscuro** [comment]
Will be fixed in #3357.
